ssh_key_manager_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+func encodeToReader(i interface{}) io.ReadCloser {
+	b := new(bytes.Buffer)
+	enc := json.NewEncoder(b)
+	enc.Encode(i)
+
+	return ioutil.NopCloser(strings.NewReader(b.String()))
+}
+
+func TestDecodeMemberList(t *testing.T) {
+	gMember1 := &GoogleMember{"member1@uw.co.uk"}
+	gMember2 := &GoogleMember{"member2@uw.co.uk"}
+	gMemberList := &GoogleMemberList{
+		Members: []GoogleMember{*gMember1, *gMember2},
+	}
+	r := encodeToReader(*gMemberList)
+
+	memberList, err := decodeMemberList(r)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if len(memberList.Members) != 2 {
+		t.Error("Unexpected MemberList", memberList.Members)
```

Comment:
Again, you should use t.Fatal instead so that you stop execution at this point and fail fast